### ISSUE TITLE
Alpha.6 [B8ce025e] Same Cart shows different Inventories

### ISSUE THREAD

**Issue by Frozen-byte** [StrangeLoopGames/EcoIssues#2305]
My View:
![image](https://user-images.githubusercontent.com/3965439/30885165-90e67ea2-a312-11e7-93c2-34d8709000ee.png)

Fanks View:
![image](https://user-images.githubusercontent.com/3965439/30885180-96ccc39e-a312-11e7-818f-f68c31c84073.png)

I also can not access the surrounding inventories. Pressing E does nothing; when pressing ESC a blank panel appears: 
![image](https://user-images.githubusercontent.com/3965439/30885254-c7291696-a312-11e7-8a6e-08b7bffd0b66.png)

**Comment by MarjieAVolk** [comment]
Do you have authorization to access the surrounding inventories?  If not, I would say this is all expected behavior, except for the blank panel.

So when you hit 'E' looking at the chest, at first you see nothing, but then when you open the escape menu, you see this blank screen pop up?  Sounds like client exceptions in ui initialization - can you attach your error log?

**Comment by Frozen-byte** [comment]
I am allowed to access the Chests.

I will attach the logs as this occurs next time. Stay tuned

**Comment by Frozen-byte** [comment]
could reproduce in player.log it spams the following error every millisecond:
```
NullReferenceException: Object reference not set to an instance of an object
  at WorldObject.Update () [0x00000] in <filename unknown>:0 
 
(Filename:  Line: -1)
```

**Comment by MarjieAVolk** [comment]
Could you attach the whole log?  That sounds like an after effect of the actual problem

**Comment by Frozen-byte** [comment]
Well the logfile grows by 200kb every second (!!!) I truncated the 30mb file into 160kb: 
[Player4.log](https://github.com/StrangeLoopGames/EcoIssues/files/1338609/Player4.log)

**Comment by Fank** [comment]
Looks like:
```
Exception: Caught exception invoking RPC ToggleUI on PlayerClient!

Argument cannot be null.
Parameter name: source

Stack:
  at System.Linq.Check.SourceAndSelector (System.Object source, System.Object selector) [0x00000] in <filename unknown>:0 
  at System.Linq.Enumerable.Select[WorldObjectComponentView,<>__AnonType4`2] (IEnumerable`1 source, System.Func`2 selector) [0x00000] in <filename unknown>:0 
  at UI.WorldObjectUI.Init (Eco.Shared.Serialization.BSONObject bsonObj) [0x00000] in <filename unknown>:0 
  at UI.UIManager.Open (System.String guiName, Eco.Shared.Serialization.BSONObject bson, UILayer layer) [0x00000] in <filename unknown>:0 
  at UI.UIManager.Toggle (System.String guiName, Eco.Shared.Serialization.BSONObject bson, UILayer layer) [0x00000] in <filename unknown>:0 
  at GenericClient`1[T].ToggleUI (System.String uiName, Eco.Shared.Serialization.BSONObject bsonObj) [0x00000] in <filename unknown>:0 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0 

  at Eco.Shared.Networking.RPCManager.TryInvoke (System.Object controller, System.String methodname, Eco.Shared.Serialization.BSONObject bsonArgs, System.Object& result) [0x00000] in <filename unknown>:0 
  at Eco.Shared.Networking.RPCManager.InvokeOn (Eco.Shared.Serialization.BSONObject bson, System.Object controller, System.String methodname) [0x00000] in <filename unknown>:0 
  at Eco.Shared.Networking.RPCManager.HandleReceiveRPC (INetClient client, Eco.Shared.Serialization.BSONObject bson) [0x00000] in <filename unknown>:0 
  at NetworkManager.ReceiveEvent (INetClient client, NetworkEvent netEvent, Eco.Shared.Serialization.BSONObject bson) [0x00000] in <filename unknown>:0 
  at Eco.Shared.Networking.NetObject.ReceiveEvent (INetClient client, NetworkEvent netEvent, Eco.Shared.Serialization.BSONObject bson) [0x00000] in <filename unknown>:0 
  at Eco.Shared.Networking.NetPort.ProcessBuffer (Int64 maxMilliseconds) [0x00000] in <filename unknown>:0 
  at Eco.Shared.Networking.NetPort.Update (Int64 maxMilliseconds) [0x00000] in <filename unknown>:0 
  at NetworkManager.Update () [0x00000] in <filename unknown>:0 
 
(Filename:  Line: -1)
```

**Comment by MarjieAVolk** [comment]
Looks like you are getting this issue, likely the root cause: https://github.com/StrangeLoopGames/EcoIssues/issues/2283
@metachronism That issue is closed, but we've still got a couple of reports of it still happening, are you sure it's fixed?

As per Fank's comment, that also looks like an after effect of this, i.e. the world object views were not fully initialized.

**Comment by metachronism** [comment]
yes, this should be resolved with 6.1
